### PR TITLE
crimson/osd/pg: do_osd_ops_execute refactor

### PR DESCRIPTION
**Pull request by Matan-B**
Preliminary refactor around `do_osd_ops_execute()` in preparation for: https://tracker.ceph.com/issues/61651.
Mainly for separating the two returned futures (`submitted` and `all_completed`).

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/crimson/osd/pg.cc [hunk 1]
```diff
@@ -1064,16 +1064,15 @@ PG::do_osd_ops(
...
-            return it->second.all_committed.get_shared_future()
-              .then([this, &version, last_complete, log_reply_fut = std::move(log_reply_fut)] {
-              if (version != eversion_t()) {
-                peering_state.complete_write(version, last_complete);
-              }
-              return log_reply_fut();
-            });
+            fut2 = it->second.all_committed.get_shared_future();
+            if (version != eversion_t()) {
+              peering_state.complete_write(version, last_complete);
```
[the remark this answers is not in the document]
Comment (Matan-B) [reply src/crimson/osd/pg.cc]: Fixes, thanks!

**Comment by Matan-B** [comment]
jenkins retest this please

**Review by athanatos — APPROVED** [review]

**Comment by Matan-B** [comment]
https://shaman.ceph.com/builds/ceph/wip-wip-matanb-crimson-do_osd_ops_execute-v3/

**Review by cyx1231st — COMMENTED** [review]
LGTM, looks the behavior is unchanged, so probably worth adding keyword "cleanup" to the PR title?

**Comment by Matan-B** [comment]
>LGTM, looks the behavior is unchanged, so probably worth adding keyword "cleanup" to the PR title?

Yes, no behavior change. "Refactor" was added to the title, I'll add cleanup label as well. Thanks!

Merging based on tests.

failures:
7441958 - (osd.1) https://tracker.ceph.com/issues/61651
7441964 - (osd.2) https://tracker.ceph.com/issues/61651
7441986 -  (osd.2) https://tracker.ceph.com/issues/61651
perf

**Pull request merged**